src/netbase.cpp

```diff
@@ -728,6 +724,19 @@ bool SetSocketNonBlocking(SOCKET& hSocket, bool fNonBlocking)
     return true;
 }
 
+bool SetSocketNoDelay(SOCKET& hSocket)
+{
+    int set = 1;
+    int rc = setsockopt(hSocket, IPPROTO_TCP, TCP_NODELAY,
+#ifdef WIN32
```

Comment:
You mean in the middle of the function call?  I agree it looks visually worse, but I like deduplicating code as much as possible.

However on longer consideration, I realized that the conversion from (const char*) to (const void*) will happen implicitly on any platform, so this ifdef isn't even needed.